In [2]:
import pandas as pd
import scipy.stats as stats
%matplotlib inline

In [3]:
dir = '../resources/data/'
# from Kaggle https://www.kaggle.com/neuromusic/avocado-prices
filename = 'avocado.csv' 

In [4]:
df = pd.read_csv(dir + filename)

In [5]:
df.dtypes

Unnamed: 0        int64
Date             object
AveragePrice    float64
Total Volume    float64
4046            float64
4225            float64
4770            float64
Total Bags      float64
Small Bags      float64
Large Bags      float64
XLarge Bags     float64
type             object
year              int64
region           object
dtype: object

**Column Definitions**
* Date - The date of the observation
* AveragePrice - the average price of a single avocado
* type - conventional or organic
* year - the year
* Region - the city or region of the observation
* Total Volume - Total number of avocados sold
* 4046 - Total number of avocados with PLU 4046 sold
* 4225 - Total number of avocados with PLU 4225 sold
* 4770 - Total number of avocados with PLU 4770 sold


In [6]:
df.head()

Unnamed: 0        Date  AveragePrice  Total Volume     4046       4225  \
0           0  2015-12-27          1.33      64236.62  1036.74   54454.85   
1           1  2015-12-20          1.35      54876.98   674.28   44638.81   
2           2  2015-12-13          0.93     118220.22   794.70  109149.67   
3           3  2015-12-06          1.08      78992.15  1132.00   71976.41   
4           4  2015-11-29          1.28      51039.60   941.48   43838.39   

     4770  Total Bags  Small Bags  Large Bags  XLarge Bags          type  \
0   48.16     8696.87     8603.62       93.25          0.0  conventional   
1   58.33     9505.56     9408.07       97.49          0.0  conventional   
2  130.50     8145.35     8042.21      103.14          0.0  conventional   
3   72.58     5811.16     5677.40      133.76          0.0  conventional   
4   75.78     6183.95     5986.26      197.69          0.0  conventional   

   year  region  
0  2015  Albany  
1  2015  Albany  
2  2015  Albany  
3  2015  Albany  
4  2015  Albany

**Question 1**: Take a look at the **region** column and the values in it. Choosing 2 of the values in **region** along with **Total Volume**, can you form a hypothesis around the mean's of the samples? (hint: you will need to use Welch's t-Test)

**Question 1a**: What is your hypothesis?

In [7]:
# print(df[(df['region'] == 'Albany')]['Total Volume'].mean())
# print(df[(df['region'] == 'TotalUS')]['Total Volume'].mean())
# print(df[(df['region'] == 'West')]['Total Volume'].mean())
Albany = df[(df['region'] == 'Albany')]
Albany_volume = Albany['Total Volume']
Atlanta = df[(df['region'] == 'Atlanta')]
Atlanta_volume = Atlanta['Total Volume']
df.groupby('region')['Total Volume'].describe()
print("Albany and Atlanta will have different mean total Volume with Albany having more.")


Albany and Atlanta will have different mean total Volume with Albany having more.


In [8]:
t, p = stats.ttest_ind(Albany_volume, Atlanta_volume, equal_var = False)

In [9]:
def degrees_of_freedom(x, y):
    return (x.var()/x.shape[0] + y.var()/y.shape[0])**2 / ((x.var()/x.shape[0])**2 / (x.shape[0]-1) + (y.var()/y.shape[0])**2 / (y.shape[0]-1))

**Question 1b**: What is your solution using Welch's t-Test? Print the t-value, p-value and degrees of freedom

In [40]:
degrees_of_freedom = degrees_of_freedom(Albany_volume, Atlanta_volume)
print('t-value: {}'.format(t))
print('p-value: {}'.format(p))
print('degrees of freedom: {}'.format(degrees_of_freedom))

t-value: -14.733725399646264
p-value: 8.337998930581766e-39
degrees of freedom: 361.1627268861998


**Question 1c**: Can you accept or reject the null hypothesis?

In [11]:
print('I can reject the null hypothesis because the p value is significantly lower then .05. Because of this we know that this is a statistically significant difference')

I can accept the null hypothesis because the p value is significantly lower then .05. Because of this we know that this is a statistically significant difference


**Question 2**: Take a look at the **AveragePrice** and **AveragePrice** columns. Economic principle stipulates as Volume goes up, Price should go down? Is there a correlation between the 2? Can you confirm the economic principle in this case?

In [15]:
abs(df['AveragePrice'].corr(df['Total Volume']))

0.19275238715271908

In [16]:
print('There is a low correlation and so I cant confirm this economic principle.')

There is a low correlation and so I cant confirm this economic principle.
